subset_vds.py

```diff
@@ -0,0 +1,373 @@
+            contig = split_interval[0]
+            start = "start"
+            end = "end"
+        elif len(split_interval) == 2:
+            contig = split_interval[0]
+            assert int(
+                start := split_interval[1]
+            ), f"If only one position is specified, it must be an int, not {start}"
+            start = int(split_interval[1])
+            end = int(split_interval[1]) + 1
+        elif len(split_interval) == 3:
+            contig = split_interval[0]
+            if start != "start":
+                assert int(
+                    start := split_interval[1]
+                ), f"The start value: {start} couldn't be converted to an int"
+                if int(start) < 1:
+                    raise ValueError(
+                        "Chromosome start positions must be greater than 0."
+                    )
+            start = int(split_interval[1])
+            if end != "end":
+                assert int(
+                    end := split_interval[2]
+                ), f"The end value: {end} couldn't be converted to an int."
+                if int(end) > get_reference("GRCh38").lengths[contig]:
+                    end = get_reference("GRCh38").lengths[contig]
+            end = int(split_interval[2])
+        else:
+            raise ValueError(
+                f"The interval: {interval} does not conform to the required format of either 'chrom', 'chrom:pos' or 'chrom:start-end'."
+            )
+        if start != "start" and end != "end":
+            assert int(start) < int(end)
+        locus_list.append(
+            parse_locus_interval(f"{contig}:{start}-{end}", reference_genome="GRCh38")
+        )
+    return locus_list
+
+
+def get_subset_sample_list(input_vds: VariantDataset, n_samples: int) -> list[str]:
+    """generates a list of samples from the input vds to use for subsetting
+
+    Args:
+        input_vds (VariantDataset): the input vds
+        n_samples (int): the number of samples to subset to
+
+    Raises:
+        ValueError: if n_samples is 0, throw an error as this would remove all data from the vds
+        ValueError: if n_samples is >= to the number of samples in the input vds throw an error, as this would not do any filtering
+
+    Returns:
+        list[str]: a random list of samples of length n_samples from the input vds, to use in subsetting
+    """
+    starting_samples: list[str] = input_vds.variant_data.s.collect()
+    n_total_samples: int = len(starting_samples)
+    if n_samples == 0:
+        raise ValueError(
+            "Subsetting to 0 samples will result in an empty MatrixTable, so not doing that."
+        )
+    if n_samples >= n_total_samples:
+        raise ValueError(
+            f"The subset sample size {n_samples} is at least the same size as the total number of samples in the input {n_total_samples}. Please choose a smaller number."
+        )
+    subset_sample_list = sample(starting_samples, n_samples)
```

Comment:
do we want to seed this for reproducibility?

Reply:
We can get this for free, so why not